pkg/providers/provider.go

```diff
@@ -66,3 +67,8 @@ type MachineConfig interface {
 	GetNamespace() string
 	GetName() string
 }
+
+// KubeconfigReader reads the kubeconfig secret on a cluster and copies the contents to a writer.
+type KubeconfigReader interface {
```

Comment:
Why do we define this in the provider package?
Ideally we would only define it in the consumer of this interface
If for some reason we need a shared definition, then the kubeconfig package sounds like the right place

Reply:
moved it to kubeconfig